test/rpc/endpoints/get_version_info_test.exs

```diff
@@ -0,0 +1,50 @@
+defmodule Soroban.RPC.CannedGetVersionInfoClientImpl do
+  @moduledoc false
+  @behaviour Soroban.RPC.Client.Spec
+
+  @impl true
+  def request(_endpoint, _url, _headers, _body, _opts) do
+    send(self(), {:request, "RESPONSE"})
+
+    {:ok,
+     %{
+       version: "21.1.0",
+       commit_hash: "fcd2f0523f04279bae4502f3e3fa00ca627e6f6a",
+       build_time_stamp: "2024-05-10T11:18:38",
+       captive_core_version: "stellar-core 21.0.0.rc2 (c6f474133738ae5f6d11b07963ca841909210273)",
+       protocol_version: 21
+     }}
+  end
+end
```

Comment:
This code is repeated twice. It was implemented this way because all the other endpoints were developed in this way, ideal to improve the code in future implementations.